### PR TITLE
Initial import

### PR DESCRIPTION
**Pull request by sagikazarmark**

### CODE REVIEW

File: src/Extractor/ProtocolVersionExtractor.php [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+
+/**
+ * @author GeLo <geloen.eric@gmail.com>
+ */
+class ProtocolVersionExtractor
+{
+    /**
+     * Extracts the protocol version
+     *
+     * @param array|string $headers
```
Comment (egeloen) [inline src/Extractor/ProtocolVersionExtractor.php]: We can replace `array` by `string[]`. Same for other extractors.

File: src/Normalizer/BodyNormalizer.php [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+
+/**
+ * @author GeLo <geloen.eric@gmail.com>
+ */
+class BodyNormalizer
+{
+    /**
+     * Normalizes the body
+     *
+     * @param mixed  $body
```
Comment (egeloen) [inline src/Normalizer/BodyNormalizer.php]: `mixed` can be replaced by `callable|string`

File: src/Normalizer/BodyNormalizer.php [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+     * @param string $method
+     *
+     * @return mixed
```
Comment (egeloen) [inline src/Normalizer/BodyNormalizer.php]: `mixed` can be replaced by `string|null`

File: src/Normalizer/HeadersNormalizer.php [hunk 4]
```diff
@@ -0,0 +1,82 @@
...
+ */
+class HeadersNormalizer
+{
+    /**
+     * Normalizes the headers
+     *
+     * @param string|array $headers
+     * @param boolean      $associative TRUE if the headers should be an associative array else FALSE.
+     *
+     * @return array
```
Comment (egeloen) [inline src/Normalizer/HeadersNormalizer.php]: `array` can be replaced by `string[]`

File: src/Parser/CookieParser.php [hunk 5]
```diff
@@ -0,0 +1,66 @@
...
+ * @author GeLo <geloen.eric@gmail.com>
+ */
+class CookieParser
+{
+    /**
+     * Parses a cookie header
+     *
+     * @param string $header
+     *
+     * @return array The parsed cookie header (0 => name, 1 => value, 2 => attributes).
```
Comment (egeloen) [inline src/Parser/CookieParser.php]: Not sure about it... What about `mixed[]`?

File: src/Parser/HeadersParser.php [hunk 6]
```diff
@@ -0,0 +1,54 @@
...
+
+/**
+ * @author GeLo <geloen.eric@gmail.com>
+ */
+class HeadersParser
+{
+    /**
+     * Parses the headers
+     *
+     * @param array|string $headers
```
Comment (egeloen) [inline src/Parser/HeadersParser.php]: `array` can be replaced by `string[]`

File: src/Parser/HeadersParser.php [hunk 7]
```diff
@@ -0,0 +1,54 @@
...
+     *
+     * @return array
```
Comment (egeloen) [inline src/Parser/HeadersParser.php]: `array` can be replaced by `string[]`

File: src/Normalizer/BodyNormalizer.php (hunk 2 above)
Comment (egeloen) [inline src/Normalizer/BodyNormalizer.php]: See https://github.com/php-http/adapter-helper/pull/1/files#diff-744a2a439d7c7ef38b3e1ce013b91bfcR21 for all the use cases...

Comment (sagikazarmark) [inline src/Normalizer/BodyNormalizer.php]: It is

File: src/Parser/CookieParser.php (hunk 5 above)
Comment (sagikazarmark) [inline src/Parser/CookieParser.php]: I think this is fine. Mixed as return type is generally a bad idea. The possible return values can be mentioned in the comment.

File: src/Normalizer/BodyNormalizer.php [hunk 8]
```diff
@@ -0,0 +1,42 @@
...
+
+/**
+ * @author GeLo <geloen.eric@gmail.com>
+ */
+class BodyNormalizer
+{
+    /**
+     * Normalizes the body
+     *
+     * @param callable|resource|string|StreamInterface $body
```
Comment (egeloen) [inline src/Normalizer/BodyNormalizer.php]: Can you add the `StreamInterface` use statement?

File: src/Normalizer/HeaderNormalizer.php [hunk 9]
```diff
@@ -0,0 +1,82 @@
...
+     */
+    public static function normalizeHeaderName($name)
+    {
+        return trim($name);
+    }
+
+    /**
+     * Normalizes the header value
+     *
+     * @param array|string $value
```
Comment (egeloen) [inline src/Normalizer/HeaderNormalizer.php]: `array` can be replaced by `string[]`

**Pull request merged**